open-reac/src/main/java/com/powsybl/openreac/parameters/input/OpenReacParameters.java

```diff
@@ -657,6 +666,161 @@ private static String getMissingLimitsMessageKey(Pair<Integer, Integer> value) {
+                                                                      branch.r, branch.x, branch.ratio);
+                LOGGER.warn("French branch with high impedance ratio: {} '{}': r={} Ω, x={} Ω (r/|x|={}) [Vnom1={} kV, Vnom2={} kV]",
+                    branch.type, branch.id, branch.r, branch.x, branch.ratio, branch.vNom1, branch.vNom2);
+            });
+        }
+
+        // Report warnings for non-French branches
+        if (!problematicNonFrenchBranches.isEmpty()) {
+            Reports.reportNbNonFrenchBranchesWithHighImpedanceRatio(reportNode, problematicNonFrenchBranches.size());
+            problematicNonFrenchBranches.forEach(branch -> {
+                Reports.reportNonFrenchBranchWithHighImpedanceRatio(reportNode, branch.id, branch.type,
+                                                                      branch.r, branch.x, branch.ratio);
+                LOGGER.warn("Non-French branch with high impedance ratio: {} '{}': r={} Ω, x={} Ω (r/|x|={}) [Vnom1={} kV, Vnom2={} kV]",
+                    branch.type, branch.id, branch.r, branch.x, branch.ratio, branch.vNom1, branch.vNom2);
+            });
+        }
+
+        // Report and throw error for French branches
+        if (!violatingFrenchBranches.isEmpty()) {
+            Reports.reportNbFrenchBranchesWithHighImpedanceRatio(reportNode, violatingFrenchBranches.size());
+            StringBuilder errorMessage = new StringBuilder();
+            errorMessage.append("The following French branches have r > 10 * |x|, which is not supported by OpenReac:\n");
+            violatingFrenchBranches.forEach(branch -> {
+                Reports.reportFrenchBranchWithHighImpedanceRatio(reportNode, branch.id, branch.type,
+                                                                   branch.r, branch.x, branch.ratio);
+                String message = String.format("%s '%s': r=%.6f Ω, x=%.6f Ω (r/|x|=%.2f) [Vnom1=%.1f kV, Vnom2=%.1f kV]",
+                    branch.type, branch.id, branch.r, branch.x, branch.ratio, branch.vNom1, branch.vNom2);
+                errorMessage.append("  - ").append(message).append("\n");
+                LOGGER.error("French branch impedance validation failed: {}", message);
+            });
+            throw new InvalidParametersException(errorMessage.toString());
+        }
+    }
+
+    /**
+     * Internal class to store branch impedance information
+     */
+    private static class BranchImpedanceInfo {
+        final String id;
+        final String type;
+        final double r;
+        final double x;
+        final double ratio;
+        final double vNom1;
+        final double vNom2;
+
+        BranchImpedanceInfo(String id, String type, double r, double x, double ratio, double vNom1, double vNom2) {
+            this.id = id;
+            this.type = type;
+            this.r = r;
+            this.x = x;
+            this.ratio = ratio;
+            this.vNom1 = vNom1;
+            this.vNom2 = vNom2;
+        }
+    }
```

Comment:
You can use a [record](https://docs.oracle.com/en/java/javase/21/language/records.html) instead:

```suggestion
    private record BranchImpedanceInfo(String id, String type, double r, double x, double ratio, double vNom1, double vNom2) {
    }
```

Reply:
Thanks for this discovery, I didn't know about it. The change has been made in the next commit.